component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java

```diff
@@ -189,12 +231,39 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
     UncertainCount[] counts = getCounts(calResult.getSize(), calArchiveResult.getSize(),
         navDetails.getOffset(), nb, query != null);
 
-    CalendarNavigation calendarNavigation = new CalendarNavigation(
+    NavigationDetails nextNavDetails = getNextNavDetails(calDocRef, navDetails, nb, query,
+        calResult);
+    NavigationDetails prevNavDetails = getPrevNavDetails(calDocRef, navDetails, nb, query,
+        calResult, calArchiveResult);
+    CalendarNavigation calendarNavigation = null;
+    calendarNavigation = new CalendarNavigation(
         counts[0], counts[1], counts[2], navDetails, startNavDetails, endNavDetails,
-        getPrevNavDetails(calDocRef, navDetails, nb, query, calResult, calArchiveResult),
-        getNextNavDetails(calDocRef, navDetails, nb, query, calResult));
-    LOGGER.debug("getCalendarNavigation: return '" + calendarNavigation + "' for cal '"
-        + calDocRef + "', navDetails '" + navDetails + "' and query '" + query + "'");
+        prevNavDetails, nextNavDetails);
+    System.out.println("<<<<<<<<<<<<<<<<<<<<<<<<<< CalendarNavigationFactory getCalendarNavigation countAfter: ["
+        + calendarNavigation.getCountAfter().getCount()
+        + "] offset: [" + navDetails.getOffset()
+        + "] nb: [" + nb + "] actualOffset: [" + actualOffset + "]");
+    System.out.println();
+    if((calendarNavigation.getCountAfter().getCount() <= 0)
+        && (navDetails.getOffset() <= 0) && (actualOffset > 0) && isSendingEmptyPage) {
+      try {
+        calendarNavigation = new CalendarNavigation(
+            new UncertainCount(0, false),
+            new UncertainCount(0, false),
+            new UncertainCount(0, false), 
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0));
+      } catch (NavigationDetailException exc) {
+        LOGGER.error("Exception getting calNav for cal [" + calDocRef + "], eventDate ["
+            + navDetails.getStartDate() + "], " + "offset [" + navDetails.getOffset()
+            + "], nb ["+ nb +"], query ["+ query +"]", exc);
+      }
```

Comment:
Extract to own method for reusability and code readability.

Also, NavigationDetailException should never happen in this case.